File: 2-second-year-2022-2023/Programming-languages-and-laboratory/Java/Exercises/08_compito_Java 2016-02-16/Professionista.java
import java.util.*;

public class Professionista extends Transazione {
	
  private String ragioneSociale;
  private int partitaIva;
  private int codice;
  
  public Professionista(int id, int giorno,
  int mese,int anno,String ragioneSociale, int partitaIva,int codice){
    super(id,giorno,mese,anno);
    this.ragioneSociale = ragioneSociale;
    this.partitaIva = partitaIva;
    this.codice=codice;
  }
  
  public String toString(){
	  return id +"\t"+giorno +"/"+mese+"/"+anno+ "\t-\t-\t-\t" 
   +ragioneSociale+"\t"+partitaIva+"\t"+codice;
  }
}

File: II - Linguaggi di programmazione e laboratorio/Esame/Python/08_compito_Java 20160216/Professionista.py
from Acquisto import Acquisto


class Professionista(Acquisto):
    def __init__(self, tipo, codice, giorno, mese, anno, ragione_sociale, partita_iva, codice_cliente):
        super().__init__(tipo, codice, giorno, mese, anno)
        self._ragione_sociale = ragione_sociale
        self._partita_iva = partita_iva
        self._codice_cliente = codice_cliente

    def __str__(self):
        return super().__str__() + "\t-\t-\t-\t" + self._ragione_sociale + "\t" + str(self._partita_iva) + "\t" + str(self._codice_cliente)
